tools/smithy-rs-tool-common/src/release_tag.rs

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+use anyhow::bail;
+use lazy_static::lazy_static;
+use regex::Regex;
+use semver::Version;
+use std::cmp::Ordering;
+use std::fmt;
+use std::str::FromStr;
+
+lazy_static! {
+    static ref VERSION_TAG: Regex = Regex::new(r"^v(\d+)\.(\d+)\.(\d+)$").unwrap();
```

Comment:
may we never make an RC tagged release 🤞

Reply:
There will be no more versioned release tags. They're all date-based from here on out.